exec/java-exec/src/main/java/org/apache/drill/exec/dotdrill/DotDrillFile.java

```diff
@@ -55,6 +56,13 @@ public DotDrillType getType(){
    * @return Return owner of the file in underlying file system.
    */
   public String getOwner() {
+    if (type == DotDrillType.VIEW && status.getOwner().isEmpty()) {
```

Comment:
What about regular tables, parquet metadata cache files or stat files? Is it possible that empty `status.getOwner()` may cause problems for them?

Reply:
No, the other 'files' work fine.

Reply:
If owner is not always empty for .view.drill files on S3, maybe it makes sense to inverse the check a little bit, like: 

```java
  public String getOwner() {
    String owner = status.getOwner();
    if (owner.isEmpty() && type == DotDrillType.VIEW) {
      // Drill view S3AFileStatus is not populated with owner (it has default value of "").
      // This empty String causes IllegalArgumentException to be thrown (if impersonation is enabled) in
      // SchemaTreeProvider#createRootSchema(String, SchemaConfigInfoProvider). To work-around the issue
      // we can return current user as if they were the owner of the file (since they have access to it).
      owner = ImpersonationUtil.getProcessUserName();
    }
    return owner;
  }
```
Also what if owner is empty but file type is not ```DotDrillType.VIEW``` ? Should we throw exception in such case to detect potential future problems early in such case ?

Reply:
Owner for views on S3 is always empty.
The `getOwner()` method is currently called for views only and for other files the method which throws the `IllegalArgumentException` won't be called with empty `String`. So I suppose we should not throw an `Exception` from the method.